### PR TITLE
Accept invitations when not logged in

### PR DESCRIPTION
**Pull request by bantic**
@mixonic for your review cc @sandersonet 

  * add a /signup/invitations/:invitation_id/:verification_code route
    * this route is for unauthenticated users that click the link they receive in the email when invited to an organization. It has a lot of shared behavior with the current sign up route.
    * extract shared behaviors for signup and signup/invitations into a controller and route mixin

  * the primary differences between signup and signup/invitations are:
    * signup creates an organization after it creates a new user
    * signup redirects to "welcome.first-app" after success
    * signup/invitations does not create an organization
    * signup/invitations redirects to claim/:invitation_id/:verification_code after success
    * signup/invitations shows the name of the organization you are signing up to join

* the primary shared behaviors between the two:
    * validations for user input (email, name, password)
    * create a new user

* change the accept invite page to show the name of the organization the user is invited to join

The auth api was updated in https://github.com/aptible/auth.aptible.com/pull/134 to allow public access to invitations. Rather than trusting the `organization_name` parameter in the URL from the invitation email, we retrieve this info from the auth api.

Also:

  * break the authentication acceptance test file into login, signup/index and signup/invitations tests

The styling and messaging of the new template(s) may need some additional work.

In addition, auth needs to be updated to store the inviter name so that we can display it to the user who is accepting the invitation.

### CODE REVIEW

**Comment by bantic** [comment]
fixes #252

File: app/claim/route.js [hunk 1]
```diff
@@ -1,20 +1,43 @@
...
+    controller.set('verificationCode', verificationCode);
+  },
+
+  afterModel(model, transition){
+    if (!this.get('session.isAuthenticated')) {
+      this.get('session').attemptedTransition = transition;
+
+      let invitationId = model.invitation.id;
+      let verificationCode = model.verificationCode;
+      this.transitionTo('signup.invitations', invitationId, verificationCode);
```
Comment (mixonic) [inline app/claim/route.js]: why transition to the id instead of the model?

File: app/router.js [hunk 2]
```diff
@@ -52,7 +52,9 @@ export default Router.map(function() {
   this.route("stacks");
   this.route("login");
   this.route("logout");
-  this.route("signup");
+  this.route("signup", {}, function(){
+    this.route('invitations', {path:'/invitations/:invitation_id/:verification_code'});
```
Comment (mixonic) [inline app/router.js]: may make more sense to make this `invitation` even if the path is `invitations`

File: app/claim/route.js (hunk 1 above)
Comment (bantic) [inline app/claim/route.js]: as in make an object with an id and verification code?
i.e.: `transitionTo('signup.invitations', {invitationId: id, verificationCode: code}` ? I have no strong preference either way

File: app/router.js (hunk 2 above)
Comment (bantic) [inline app/router.js]: sure, that makes sense to me too. I'll fold this in with any other changes

File: app/claim/route.js (hunk 1 above)
Comment (mixonic) [inline app/claim/route.js]: Oof, I see. You need access to `params` on that route, which means `model` hook.

File: app/signup/invitations/route.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    let invitationId = params.invitation_id;
+    let verificationCode = params.verification_code;
+
+    return this.store.find('invitation', invitationId).then( (invitation) => {
+      return { invitation, verificationCode };
+    });
+  },
+
+  setupController: function(controller, model){
+    let {invitation, verificationCode} = model;
```
Comment (mixonic) [inline app/signup/invitations/route.js]: I suggest adding an assertion that `model` is not an invitation instance. It seems easy to trip yourself up on.

The model-hook-skipping behavior really sucks here. IMO assertion is fine, the only other idea I can think of is reading params directly off `transition` inside `beforeModel`.

**Comment by bantic** [comment]
@mixonic addressed your comments. I'll plan on merging this after tests are green unless you have any objections

**Pull request merged**